src/cryptonote_core/tx_pool.cpp

```diff
@@ -1137,14 +1241,15 @@ namespace cryptonote
     return stake ? stake->amount >= config::graft::TIER1_STAKE_AMOUNT : false;
   };
 
-  bool tx_memory_pool::belongs_to_auth_sample(const rta_header& rta_hdr) const
+  bool tx_memory_pool::belongs_to_auth_sample(const rta_header& rta_hdr, const uint32_t sn_pkeys_off) const
```

Comment:
I would change it so it accepts two sets of public keys - one set from tx and another set is auth sample. This way it doesn't need to be a `tx_memory_pool` member and can be covered with unit test

Reply:
Will do, thanks.

Reply:
Fixed